project-koku/koku
# add instance_type key from AdditionalInfo to AzureCostEntryProductService

#1398 
This PR adds an `instance_type` key to the AzureCostEntryProductService table.

Testing:
[csv_files.tar.gz](https://github.com/project-koku/koku/files/3945147/csv_files.tar.gz)
1. Using attached data, add 4 azure-local providers and ingest data. This json can be used in Postman to create a provider (update the export_name and name fields accordingly):
```
{
    "name": "Test Azure Source - 2",
    "type": "AZURE-local",
    "authentication": {
        "credentials": {
            "subscription_id": "12345678-1234-5678-1234-567812345678",
            "tenant_id": "12345678-1234-5678-1234-567812345678",
            "client_id": "12345678-1234-5678-1234-567812345678",
            "client_secret": "12345"
        }
    }, "billing_source": {
        "data_source": {
            "resource_group": {
                "directory": "",
                "export_name": "hccmnise02"
            },
            "storage_account": {
                "local_dir": "/tmp/local_container",
                "container": ""
            }
        }
    }
}
```
2. Query the Azure data: i.e. `?filter[resolution]=daily&filter[time_scope_value]=-2&group_by[instance_type]=*`
3. See that `no-instance-type`, `Standard_A0`, and `Standard_B2` show for each day and correspond to the appropriate csv.

Results:
[1398_db.txt](https://github.com/project-koku/koku/files/3945273/1398_db.txt)

Additional Testing:
1. Run iqe smoke tests.
